PWGCF/TableProducer/filter2Prong.cxx

```diff
@@ -0,0 +1,74 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/AnalysisDataModel.h"
+#include "Framework/ASoAHelpers.h"
+
+#include "MathUtils/detail/TypeTruncation.h"
+
+#include "PWGCF/DataModel/CorrelationsDerived.h"
+
+#include "PWGHF/Core/HfHelper.h"
+#include "PWGHF/DataModel/CandidateReconstructionTables.h"
+#include "PWGHF/DataModel/CandidateSelectionTables.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+using namespace o2::math_utils::detail;
+
+#define FLOAT_PRECISION 0xFFFFFFF0
+#define O2_DEFINE_CONFIGURABLE(NAME, TYPE, DEFAULT, HELP) Configurable<TYPE> NAME{#NAME, DEFAULT, HELP};
+
+struct Filter2Prong {
+  O2_DEFINE_CONFIGURABLE(cfgVerbosity, int, 0, "Verbosity level (0 = major, 1 = per collision)")
+
+  Produces<aod::CF2ProngTracks> output2ProngTracks;
+
+  using HFCandidates = soa::Join<aod::HfCand2Prong, aod::HfSelD0>;
+  void processData(aod::Collisions::iterator const& collision, aod::BCsWithTimestamps const&, aod::CFCollRefs const& cfcollisions, aod::CFTrackRefs const& cftracks, HFCandidates const& candidates)
+  {
+    if (cftracks.size() <= 0)
+      return; // rejected collision
+    if (cfgVerbosity > 0 && candidates.size() > 0)
+      LOGF(info, "Candidates for collision: %lu, cfcollisions: %lu, CFTracks: %lu", candidates.size(), cfcollisions.size(), cftracks.size());
+    for (auto& c : candidates) {
+      int prongCFId[2] = {-1, -1};
+      for (auto& cftrack : cftracks) {
+        if (c.prong0Id() == cftrack.trackId()) {
+          prongCFId[0] = cftrack.globalIndex();
+          break;
+        }
+      }
+      for (auto& cftrack : cftracks) {
+        if (c.prong1Id() == cftrack.trackId()) {
+          prongCFId[1] = cftrack.globalIndex();
+          break;
+        }
+      }
+      // look-up the collision id
+      auto collisionId = cfcollisions.begin().globalIndex();
```

Comment:
Could cfcollisions have 0 length?

Reply:
That should always be 1. On line 40 I check already if tracks is 0 but I could check both there?

Reply:
This could skip, couldn't it: https://github.com/AliceO2Group/O2Physics/blob/ac5086c65511ddc02ef6726bb1cd5d9fd824bb5a/PWGCF/TableProducer/filterCorrelations.cxx#L100

Reply:
Yes, although then it should also skip all the tracks. But I'm adding the collision size check too just in case.